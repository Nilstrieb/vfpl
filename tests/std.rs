mod common;

vfpl_run_test!(
    print_different_types,
    r#"
    
    please create function void with no parameters that returns absent
        please return absent from the function.
    please end function void.
    
    please call println with the argument True as x.
    please call println with the argument False as x.
    please call println with the argument 532 as x.
    please call println with the argument -5532 as x.
    please call println with the argument -3245.435 as x.
    please call println with the argument 0.435 as x.
    please call println with the argument "test" as x.
    please call println with the argument absent as x.
    please call println with the argument null as x.
    please call println with the argument undefined as x.
    please call println with the argument novalue as x.
    please call println with the argument println as x.
    please call println with the argument void as x.
    
    please go to sleep.
    "#
);

vfpl_run_test!(
    time_without_error,
    "
    please initialize variable our_time as Integer with the value of (call time with no arguments).
    <!-- we cannot print the value here, we have to trust that it returned the correct time. -->
    please go to sleep.
    "
);
