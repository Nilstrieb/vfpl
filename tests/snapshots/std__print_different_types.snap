---
source: tests/std.rs
expression: stderr

---
true
false
532
-5532
-3245.435
0.435
test
absent
null
undefined
novalue
[native function]
[function]

